File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.Stack;

class Solution {
    boolean solution(String s) {
        Stack<String> stack = new Stack<>();
        boolean answer = true;

        for (int i = 0; i < s.length(); i++) {
            if (s.substring(i, i + 1).equals("(")) {
                stack.push(s.substring(i, i + 1));
            } else {
                if (stack.isEmpty()) {
                    return false;
                }
                stack.pop();
            }
        }
        if (stack.isEmpty()) {
            answer = true;
        } else {
            answer = false;
        }
        return answer;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True

    stack=[]

    for i in range(0, len(s)):
        word = s[i : i+1]
        if word=='(':
            stack.append('(')
        else:
            if(len(stack)==0):
                return False
            stack.pop()

    if(len(stack)>0):
        return False

    return True